examples/src/bin/custom_editable.rs

```diff
@@ -0,0 +1,434 @@
+    border-spacing: 4px;
+  }
+  tag label,
+  tag image {
+    color: white;
+  }
+  tag button {
+    min-height: 0;
+    min-width: 0;
+    padding: 0;
+    border: 1px solid white;
+  }
+  
+  entry.tagged {
+    border-spacing: 4px;
+  }
+";
+
+mod imp {
+    use super::*;
+    use glib::subclass::Signal;
+    use glib::ParamSpec;
+    use once_cell::sync::Lazy;
+    use std::cell::{Cell, RefCell};
+
+    pub struct CustomEditable {
+        pub text: gtk::Text,
+        pub spinner: RefCell<Option<gtk::Spinner>>,
+        pub show_spinner: Cell<bool>,
+    }
+
+    impl Default for CustomEditable {
+        fn default() -> Self {
+            Self {
+                text: gtk::Text::new(),
+                spinner: RefCell::default(),
+                show_spinner: Cell::new(false),
+            }
+        }
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for CustomEditable {
+        const NAME: &'static str = "CustomEditable";
+        type Type = super::CustomEditable;
+        type ParentType = gtk::Widget;
+        type Interfaces = (gtk::Editable,);
+
+        fn class_init(klass: &mut Self::Class) {
+            klass.set_layout_manager_type::<gtk::BoxLayout>();
+            klass.set_css_name("entry");
+            klass.set_accessible_role(gtk::AccessibleRole::TextBox);
+        }
+    }
+
+    impl ObjectImpl for CustomEditable {
+        fn properties() -> &'static [ParamSpec] {
+            static PROPERTIES: Lazy<Vec<ParamSpec>> = Lazy::new(|| {
+                vec![ParamSpec::boolean(
+                    "show-spinner",
+                    "Spinner shown",
+                    "Whether the editable has a visible spinner",
+                    false,
+                    glib::ParamFlags::READWRITE,
+                )]
+            });
+            PROPERTIES.as_ref()
+        }
+
+        fn set_property(
+            &self,
+            editable: &Self::Type,
+            id: usize,
+            value: &glib::Value,
+            pspec: &glib::ParamSpec,
+        ) {
+            if !self.delegate_set_property(editable, id, value, pspec) {
+                match pspec.get_name() {
+                    "show-spinner" => {
+                        editable.set_show_spinner(value.get_some().unwrap());
+                    }
+                    _ => unimplemented!(),
+                }
+            }
+        }
+
+        fn get_property(
+            &self,
+            editable: &Self::Type,
+            id: usize,
+            pspec: &glib::ParamSpec,
+        ) -> glib::Value {
+            if let Some(value) = self.delegate_get_property(editable, id, pspec) {
+                value
+            } else {
+                match pspec.get_name() {
+                    "show-spinner" => self.show_spinner.get().to_value(),
+                    _ => unimplemented!(),
+                }
+            }
+        }
+
+        fn constructed(&self, editable: &Self::Type) {
+            editable.init_delegate();
```

Comment:
Some comments about these things would be useful

Reply:
Added more details & a bit of explanations